serve.py

```diff
@@ -55,13 +59,33 @@ def get_miner_positions():
     if api_key not in accessible_api_keys:
         return jsonify({'error': 'Unauthorized access'}), 401
 
-    f = "outputs/output.json"
-    data = get_file(f)
+    # Get the 'tier' query parameter from the request
+    tier = request.args.get('tier')
+    is_gz_data = tier is not None
+
+    if is_gz_data:
+        # Validate the 'tier' parameter
+        if tier not in ['0', '30', '50', '100']:
+            return jsonify({'error': 'Invalid tier value. Allowed values are 0, 30, 50, or 100'}), 400
+
+        # Construct the relative path based on the specified tier
+        f = f"outputs/tiered_positions/{tier}/output.json.gz"
+    else:
+        # If 'tier' parameter is not provided, return the default output.json
+        f = "outputs/output.json"
+
+    # Attempt to retrieve the file
+    data = get_file(f, binary=is_gz_data)
 
     if data is None:
         return f"{f} not found", 404
+    if is_gz_data:
+        return Response(data, content_type='application/json', headers={
+            'Content-Encoding': 'gzip'
+        })
     else:
```

Comment:
This else statement is not needed because of the return.

Reply:
good catch! fixed